airflow/providers/apache/hive/hooks/hive.py

```diff
@@ -775,6 +775,20 @@ def table_exists(self, table_name, db='default'):
         except Exception:  # pylint: disable=broad-except
             return False
 
+    def drop_partitions(self, table_name, part_vals, delete_data=False, db='default'):
+        """
+        Drop partitions matching param_names input
+        >>> hh = HiveMetastoreHook()
+        >>> hh.drop_partitions(db='airflow', table_name='static_babynames', part_vals="['2020-05-01']")
+        True
+        """
+        if self.table_exists(table_name, db):
+            with self.metastore as client:
+                return client.drop_partition(db, table_name, part_vals, delete_data)
+        else:
+            self.log.info("Table %s.%s does not exist!" % (db, table_name))
```

Comment:
maybe make a log message about the partition existing. do we want to check that too before we try to drop the partition? There is a function in the hook for this too.

Reply:
Got it. i can add that.